cybox/common/vocabs.py

```diff
@@ -33,21 +33,41 @@ def __init__(self, *args, **kwargs):
         # should always be a subclass of VocabString.
         self.type_ = VocabString  # Force this
 
-    def __set__(self, instance, value):
-        """Overrides cybox.TypedField.__set__()."""
+    def _handle_value(self, value):
+        """Handles the processing of the __set__ value.
+
+        1) If the value is ``None``, return ``None``
+        2) If the value is an instance of ``VocabString``, return it.
+        3) Attempt to cast the value to the default VocabString type if there
+           is one, else try to cast it to VocabString.
+        4) raise a ValueError
+
+        """
         vocab = self.__vocab_impl
 
         if value is None:
-            instance._fields[self.name] = None
+            return None
         elif isinstance(value, VocabString):
-            instance._fields[self.name] = value
+            return value
         elif vocab._try_cast:  # noqa
-            value = vocab(value)
-            instance._fields[self.name] = value
+            return vocab(value)
+
+        error_fmt = "%s must be a %s, not a %s"
+        error = error_fmt % (self.name, self.type_, type(value))
+        raise ValueError(error)
+
+    def __set__(self, instance, value):
```

Comment:
Any chance we can try to consolidate this behavior with `TypedField.__set__`, so that the only difference between `TypedField` and `VocabField` is the implementation of `_handle_value`?